src/sample/combinednr.jl

```diff
@@ -370,3 +370,11 @@ end
 function Base.length(nr::CombinedNextReaction)
     return length(nr.transition_entry)
 end
+
+function Base.haskey(nr::CombinedNextReaction{K,T}, clock) where {K,T}
+    if clock isa K
+        haskey(nr.transition_entry, clock)
+    else
+        return false
+    end
+end
```

Comment:
The C language has a specific rule that every code segment should end in a newline. I hadn't thought about it for Julia. Just did it reflexively. Clearly it works fine.